### PR TITLE
Add BearerTokenLoggingFilter as a ContainerRequestFilter

### PR DESCRIPTION
**Pull request by diogoholanda**
I am still going to write tests. I just wanted to get agreement on the approach here before

### CODE REVIEW

**Comment by diogoholanda** [comment]
@markelliot @j-baker

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        } catch (Throwable t) {
+            log.debug("Unable to process auth header.", t);
+        }
+    }
+
+    private String getRawAuthHeader(ContainerRequestContext requestContext) {
+        String rawAuthHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
+        return rawAuthHeader != null
+                ? rawAuthHeader
+                : requestContext.getHeaderString(HttpHeaders.AUTHORIZATION.toLowerCase());
```
Comment (markelliot) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: jersey's header map is case-insensitive

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        MDC.remove(key);
+        requestContext.removeProperty(getRequestPropertyKey(key));
+    }
+
+    private void setUnverifiedContext(ContainerRequestContext requestContext, String key, String value) {
+        // * indicates unverified
+        setContext(requestContext, key, value + "*");
+    }
+
+    private void setContext(ContainerRequestContext requestContext, String key, String value) {
```
Comment (markelliot) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: i think it'd be clearer to inline this method

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    private void clearContext(ContainerRequestContext requestContext) {
+        clearContext(requestContext, USER_ID_KEY);
+        clearContext(requestContext, SESSION_ID_KEY);
+        clearContext(requestContext, TOKEN_ID_KEY);
+    }
+
+    private void clearContext(ContainerRequestContext requestContext, String key) {
+        MDC.remove(key);
+        requestContext.removeProperty(getRequestPropertyKey(key));
```
Comment (markelliot) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: ContainerRequestContext is per-request and not ThreadLocal state, I think this remove operation is unnecessary.

**Comment by diogoholanda** [comment]
@markelliot PR updated

**Pull request merged**